### ISSUE TITLE
ol.sphere.getLength(line) returning half of actual of distance 

### ISSUE THREAD

**Issue by pavank61295** [openlayers/openlayers#13829]
**Describe the bug**
ol.sphere.getLength(line) returning half of actual distance of two points, where **line** is Line String object which has two points in EPSG:3857 format

**To Reproduce**
1. Create LineString object with two poitns 
    line_data = [["8649887.3523805","1459052.94032918"],["8649890.62520487","1459088.71307292"]]
    line = new ol.geom.LineString([line_data])
2. Get line distance
    d =  ol.sphere.getLength(line)
3. Actual distance is ~72 m in map. but **d** is ~36m

**Expected behavior**
   d sohuld be ~72 m

**Comment by mike-000** [comment]
Distances measured in EPSG:3857 projection units are only true at the equator.  Elsewhere objects appear bigger in the flat projection than they are on the spherical planet (where Greenland is much smaller than Africa).  So `ol.sphere.getLength(line)` will return the true length, which will be smaller than `line.getLength()`.

**Comment by pavank61295** [comment]
Hi Mike,

Our requirement is to measure distance between two points. Max range is 300 m in our application. I tried 'EPSG:4326' also but still getting incorrect distance. 
Could you suggest a best way to find accurate distance. 1% error is acceptable for our requirement

**Comment by M393** [comment]
At just 13°N the error shouldn't be this large.
How do you know the actual distance is 72 m?

**Comment by pavank61295** [comment]
Mike, 
There are two known objects on map and distance between those two objects is 72m. We measured offline.
We have taken refence code from below link and made a change to show in meters instead of kilo meters. While drawing line text showing correct distance but we are getting start and end points of line. If we calculate distance between those, it's exctly half of what it it showing in text (actual distance).

https://openlayers.org/en/latest/examples/measure-style.html

**Comment by mike-000** [comment]
With your data `line.getLength()` returns 35.922

That should correspond to `Math.sqrt(dx * dx + dy * dy)` where
`dx = 8649890.62520487 - 8649887.3523805` (3.27282437)
`dy = 1459088.71307292 - 1459052.94032918` (35.77274374)

`ol.sphere.getLength(line)` returns 34.963 - only slightly smaller as it is quite close to the equator.

You would need to be at 60 degrees latitude for the spherical method to return half the flat projection distance.

**Comment by pavank61295** [comment]
But actual distance is 72 m.

**Comment by pavank61295** [comment]
Yes, 
This example using ol/sphere.getLength(), and example sending line object which has 3 points P1, P2, P1. I checked sphere.js file and it calulating line below.
>> Line objects carrying 3 points [P1, P2, P3]
>> d1 = getDistance(P1, P2)
>> d2 = getDistance(P2, P1)
>> return (d1 + d2)

**Comment by mike-000** [comment]
Yes a line `[P1, P2, P1]` will measure the distance from P1 to P2 and back to P1.  If you only need P1 to P2  use `[P1, P2]`

**Comment by pavank61295** [comment]
when I calulate length of line on map, getting correct distance.  Here **line_obj** carrying 3 points which are [P1, P2, P1]. I observed this in open layers example as I mentioned on previous comment.

line = new ol.geom.LineString(line_obj)
d = ol.sphere.getLength(line)

But If I create Line object using P1, P2 then length is half of expected.

So do we need to multiply result with 2. If yes then I'll get correct distance. But what the logic behind it?

**Comment by ahocevar** [comment]
@pavank61295 A few lines of code say more than several comments of prose. Why don't you share your code, so we can see what you're doing?

**Comment by pavank61295** [comment]
This is the code. It takes two points in 'EPSG:3857' format.

```
function get_distance(c1, c2)
{
    var d = 0
    c1 = ol.proj.transform(c1, 'EPSG:3857', 'EPSG:4326')
    c2 = ol.proj.transform(c2, 'EPSG:3857', 'EPSG:4326')

    const radius = 6371008.8;
    const lat1 = toRadians(c1[1]);
    const lat2 = toRadians(c2[1]);
    const deltaLatBy2 = (lat2 - lat1) / 2;
    const deltaLonBy2 = toRadians(c2[0] - c1[0]) / 2;
    const a =
      Math.sin(deltaLatBy2) * Math.sin(deltaLatBy2) +
      Math.sin(deltaLonBy2) *
        Math.sin(deltaLonBy2) *
        Math.cos(lat1) *
        Math.cos(lat2);
    d =  2 * radius * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
    return d
}

function toRadians(angleInDegrees) {
    return (angleInDegrees * Math.PI) / 180;
}
```
